packages/host/tests/helpers/index.gts

```diff
@@ -670,7 +683,66 @@ export function diff(
   };
 }
 
-export class MockResponse extends Response {
+function isCardSourceFetch(request: Request) {
+  return (
+    request.method === 'GET' &&
+    request.headers.get('Accept') === SupportedMimeType.CardSource &&
+    request.url.includes(testRealmURL)
+  );
+}
+
+export async function sourceFetchReturnUrlHandle(
+  request: Request,
+  defaultHandle: (req: Request) => Promise<Response | null>,
+) {
+  if (isCardSourceFetch(request)) {
+    let r = await defaultHandle(request);
+    if (r) {
+      return new MockRedirectedResponse(r.body, r, request.url) as Response;
+    }
+  }
+  return null;
+}
+
+export async function sourceFetchRedirectHandle(
+  request: Request,
+  adapter: RealmAdapter,
+) {
+  let urlParts = request.url.split('.');
+  if (
+    isCardSourceFetch(request) &&
+    urlParts.length === 1 //has no extension
+  ) {
+    const realmPaths = new RealmPaths(testRealmURL);
```

Comment:
`testRealmPathURL` is just the default test realm URL. There is an option to configure the test realm with a different URL and we do that for a few tests—don’t assume this is the realm URL.

Reply:
hmmm ok fixed this. But I made the handler pass an argument (required) of realmURL. I thought ideally, I would build something higher order which allows the handlers to have realmURL,adapter and maybeHandle available but the change seemed a bit too much and made the handlers being passed more unclear. I prefer this way of forcing the test maker to specify the realmURL.